torchaudio/csrc/ffmpeg/ffmpeg.cpp

```diff
@@ -83,19 +85,11 @@ AVFormatContext* get_format_context(
     throw std::runtime_error(
         "Failed to open the input \"" + src + "\" (" + av_err2string(ret) +
         ").");
-  return pFormat;
+  return AVFormatContextPtr(pFormat);
 }
-} // namespace
 
-AVFormatContextPtr::AVFormatContextPtr(
-    const std::string& src,
-    const std::string& device,
-    const std::map<std::string, std::string>& option)
-    : Wrapper<AVFormatContext, AVFormatContextDeleter>(
-          get_format_context(src, device, option)) {
-  if (avformat_find_stream_info(ptr.get(), NULL) < 0)
-    throw std::runtime_error("Failed to find stream information.");
-}
+AVFormatContextPtr::AVFormatContextPtr(AVFormatContext* p)
+    : Wrapper<AVFormatContext, AVFormatContextDeleter>(p) {}
```

Comment:
`AVFormatContextPtr` is only responsible for pointer clean-up. The initialization logic has been moved to call site.